### PR TITLE
[ODS-4493] Migration utility update descriptor package 

### PR DESCRIPTION
**Pull request by semalaiappan**

### CODE REVIEW

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 [hunk 1]
```diff
@@ -0,0 +1,148 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+$ErrorActionPreference = "Stop"
+$repositoryNames = @('EdFi.Ods.Utilities.Migration')
+& "$PSScriptRoot\..\..\..\logistics\scripts\modules\load-path-resolver.ps1"  $repositoryNames
```
Comment (jamessmckay) [inline EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: i am not sure we should keep logistics in the path. should we consider scripts off the root instead?

Comment (jamessmckay) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: is this path resolver referencing the ods project?

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 [hunk 2]
```diff
@@ -0,0 +1,148 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+$ErrorActionPreference = "Stop"
+$repositoryNames = @('EdFi.Ods.Utilities.Migration')
+& "$PSScriptRoot\..\..\..\logistics\scripts\modules\load-path-resolver.ps1"  $repositoryNames 
+Import-Module -Force -Scope Global (Get-RepositoryResolvedPath 'logistics\scripts\modules\tasks\TaskHelper.psm1')
```
Comment (jamessmckay) [inline EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: do we really need to use path resolving here? this is an independent repo

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/load-path-resolver.ps1 [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+param(
+    [string[]] $repositoryNames
+)
+
+if ((Get-Module | Where-Object -Property Name -eq 'path-resolver')) { return }
+
+Import-Module -Force -Scope Global "$PSScriptRoot\path-resolver.psm1" -ArgumentList @(, $repositoryNames)
```
Comment (jamessmckay) [inline EdFi.Ods.Utilities.Migration/logistics/scripts/modules/load-path-resolver.ps1]: again not sure want to use path resolver.. any reason to?

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/path-resolver.psm1 [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+    throw ($errorMsg)
+}
+
+
+#Sets a global folders variable. Is the primary access mechanism for resolving paths. Utilizes a delegate.
+if (-not ($script:folders)) {
+    $script:folders = @{ }
+    #This is used in remote web deployments to find this path resolver and CANNOT be a delegate.
+    $folders.core = Get-CorePath
+    $folders.scripts = [System.Func[Object, Object]] { return (Get-RepositoryResolvedPath "logistics\scripts\$($args[0])") }
```
Comment (jamessmckay) [inline EdFi.Ods.Utilities.Migration/logistics/scripts/modules/path-resolver.psm1]: if we keep path resolving, then lets remove the logistics in the folder structure

**Review by jamessmckay — CHANGES_REQUESTED** [review]

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+            $script:result += Invoke-Task -name $task -task { & $task $config }
+        }
+    }
+
+    Test-Error
+
+    $script:result += New-TaskResult -name '-' -duration '-'
+    $script:result += New-TaskResult -name $MyInvocation.MyCommand.Name -duration $elapsed.format
+
+    return $script:result | Format-Table
```
Comment (tmeador-dl) [inline EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: should be able to remove the TaskHelper.psm1 specific functions here: `Write-HashtableInfo`, `Use-StopWatch`, `Invoke-Task`, `New-TaskResult` ect.. Some of this is nice to have when it's easily available but it's not necessary for the functionality of this script.

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+        $xml.Save($nuspecPath)
+    }
+
+    Add-Files $nuspecPath
+    Add-ContentFiles $nuspecPath
+
+    Write-Host "Nuspec Created: " -NoNewline
+    Write-Host (Get-ChildItem $nuspecPath).FullName -ForegroundColor Green
+    Write-Host
+}
```
Comment (tmeador-dl) [inline EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: I think `New-Nuspec` is the only functionality coming from package.psm1 here, we can probably just create a .nuspec file and check it in to break this dependency

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 (hunk 2 above)
Comment (tbmeador-dlp) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1:9]: I think we should be able to remove the path-resolver, TaskHelper, and packaging modules dependencies and not duplicate them here in this repo. If we also create a .nuspec file all this script is doing is copying the descriptors into the right folder structure for the nuget package, so it shouldn't be a lot of code.

**Review by tmeador-dl — CHANGES_REQUESTED** [review]

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 (hunk 1 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: we have already scripts folder  for migration ,So it kept with logistics folder to separate it out .

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 (hunk 2 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1:7]: no . I removed it . Thanks

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 (hunk 6 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: yes , I did it  . Thank you !  EdFi.Standard.Descriptors.nuspec added

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 (hunk 2 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: No,  Remove it . Thanks

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/load-path-resolver.ps1 (hunk 3 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/load-path-resolver.ps1]: No  I remove it . Thanks

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/path-resolver.psm1 (hunk 4 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/path-resolver.psm1]: I remove it . Thanks

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 (hunk 5 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: These functions are required for build-edfi-standard-descriptors.psm1

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+            $script:result += Invoke-Task -name $task -task { & $task $config }
+        }
+    }
+
+    Test-Error
+
+    $script:result += New-TaskResult -name '-' -duration '-'
+    $script:result += New-TaskResult -name $MyInvocation.MyCommand.Name -duration $elapsed.format
+
+    return $script:result | Format-Table
```
Comment (tmeador-dl) [inline EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: > These functions are required for build-edfi-standard-descriptors.psm1

what's preventing them from being removed? would something like this work?
```suggestion
    Remove-TempDirectory $config
    New-TempDirectory $config
    Copy-PackageFiles $config
    Copy-PackageNuspec $config
```

**Review by tmeador-dl — CHANGES_REQUESTED** [review]

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 (hunk 7 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: Yes I changed it . Thanks

File: EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1 (hunk 1 above)
Comment (semalaiappan) [reply EdFi.Ods.Utilities.Migration/logistics/scripts/modules/build-edfi-standard-descriptors.psm1]: I moved to EdFi.Ods.Utilities.Migration\EdFi.Standard.Descriptors folder

**Review by jamessmckay — APPROVED** [review]

**Review by tmeador-dl — APPROVED** [review]

**Pull request merged**